pilot/pkg/bootstrap/server.go

```diff
@@ -422,7 +422,8 @@ func (s *Server) initSDSServer(args *PilotArgs) {
 			// Make sure we have security
 			log.Warnf("skipping Kubernetes credential reader; PILOT_ENABLE_XDS_IDENTITY_CHECK must be set to true for this feature.")
 		} else {
-			sc := kubesecrets.NewMulticluster(s.kubeClient, s.clusterID, args.RegistryOptions.ClusterRegistriesNamespace)
+			// TODO move this to a startup function and pass stop
```

Comment:
Is there a reason not to/additional effort needed to wrap this in `s.addStartFunc(...)` now?

Reply:
the main thing is we need to ensure it starts before serving XDS